server/server.go

```diff
@@ -1578,6 +1583,13 @@ func (s *StanServer) start(runningState State) error {
 		// Use recovered clustering node ID.
 		s.opts.Clustering.NodeID = s.info.NodeID
 
+		// Since we've recovered the clustering node ID, we need to update the
```

Comment:
This doesn't seem clearer unless I misunderstand. If there is no recovered state, we still need to assign a node id to a nuid if one is not provided, so this would mean assigning it in two different places.

Reply:
No because we said that we do not support recovering a state in cluster mode without the NodeID.  So when you recover a state in cluster mode, there will be a NodeID (so no need to set it). Check line 1580 will return an error if state is recovered without it.

Reply:
What if you're starting a server for the first time? There's no recovered state in this case, right?

Reply:
Right, so you would be in the "else" where you would set the NodeID and path and this would then be persisted (`callStoreInit = true`). Point is that this should be the only place where it is set since after that, we recover from storage (unless you are saying that one can change after the fact, which I don't think that is the case).

Reply:
I see. I think it makes sense now, thanks. :)